schema/hardware-metadata.schema.json

```diff
@@ -0,0 +1,261 @@
+{
+  "$id": "https://zmkfirmware.dev/zmk.metadata.json",
+  "title": "HardwareMetadata",
+  "$schema": "https://json-schema.org/draft/2020-12/schema",
+  "oneOf": [
+    {
+      "$ref": "#/$defs/board"
+    },
+    {
+      "$ref": "#/$defs/shield"
+    },
+    {
+      "$ref": "#/$defs/interconnect"
+    }
+  ],
+  "$defs": {
+    "id": {
+      "type": "string",
+      "pattern": "^[a-z0-9_]+$"
+    },
+    "keyboard_siblings": {
+      "type": "array",
+      "items": {
+        "type": "string"
+      }
+    },
+    "variant": {
+      "oneOf": [
+        {
+          "type": "string"
+        },
+        {
+          "type": "object",
+          "required": [
+            "id",
+            "features"
+          ],
+          "properties": {
+            "id": {
+              "$ref": "#/$defs/id"
+            },
+            "features": {
+              "$ref": "#/$defs/features"
+            }
+          }
+        }
+      ]
+    },
+    "features": {
+      "type": "array",
+      "items": {
+        "type": "string",
+        "enum": [
+          "keys",
+          "display",
+          "encoder",
+          "underglow",
+          "pointer"
+        ]
+      }
+    },
+    "interconnects": {
+      "type": "array",
+      "minItems": 1,
+      "items": {
+        "$ref": "#/$defs/id"
+      }
+    },
+    "sibling_details": {
+      "type": "object",
+      "additionalProperties": false,
+      "properties": {
+        "id": {
+          "$ref": "#/$defs/id"
+        },
+        "features": {
+          "$ref": "#/$defs/features"
+        },
+        "variants": {
+          "type": "array",
+          "items": {
+            "$ref": "#/$defs/variant"
+          }
+        }
+      }
+    },
+    "interconnect": {
+      "title": "Interconnect",
+      "type": "object",
+      "additionalProperties": false,
+      "required": [
+        "file_format",
+        "id",
+        "name",
+        "url",
+        "type"
+      ],
+      "properties": {
+        "file_format": {
+          "type": "string",
+          "const": "1"
```

Comment:
If the intent is to increment this any time we make a change to the schema, would a number be better? Then we'd be able to specify a range of allowed versions if that became useful.

Reply:
I was modeling this off of other similar schemas like the docker-compose YAML files, for instance, which use strings so they can do semantic versioning w/ minor versions that don't break compat.

Reply:
That's reasonable. I think the fact that you just used "1" instead of something like "1.0.0" made me think this was just an integer version and not a semantic version.